cpp/src/arrow/acero/asof_join_node.cc

```diff
@@ -668,18 +663,19 @@ class InputState {
 
   static Result<std::unique_ptr<InputState>> Make(
       size_t index, TolType tolerance, bool must_hash, bool may_rehash,
-      KeyHasher* key_hasher, ExecNode* node, AsofJoinNode* output,
+      KeyHasher* key_hasher, ExecNode* asof_input, AsofJoinNode* asof_node,
```

Comment:
Can you other functions in this file consistent with this naming here? e.g.
https://github.com/apache/arrow/blob/main/cpp/src/arrow/acero/asof_join_node.cc#L541

Reply:
The location you are pointing to has a different perspective of the input and output, and should not be changed. Also note that the class there could easily be pulled out for reuse, and should not be tied to specific internals of as-of-join node.

Reply:
Can you elaborate why input and output perspective are different?

Reply:
We discussed this [here](https://github.com/apache/arrow/pull/35874#discussion_r1213557505).

Reply:
Not sure I fully understand, but from my understanding both the InputState and BackpressureController takes two node, one node presents the input table of asof join, and another is the asof join node itself. So my point here is it's better to make them consistent naming to avoid confusion. And you seem to say that InputState and BackpressureController have different perspective of input and output, which I do not fully get but perhaps I misunderstood what you mean.

Reply:
(This is not a major point but I want to make sure I understand what you are saying)

Reply:
We are looking at two nodes: a node pushing to an as-of-join node. The code here has the perspective of the latter node, so the former node is seen as an input, which is why the code has `asof_input`. The `BackpressureController` has the perspective of the former node, which it controls, so the latter node is seen as an output. `BackpressureController` is designed to be reusable outside the context of `AsofJoinNode` (even though currently it is not exposed).